### ISSUE TITLE
Export log statistics

### ISSUE THREAD

**Issue by BavoKempen** [DrCoffey/DeepSqueak#22]
Dear,

When trying to export the Excel log (log statistics) from detections files obtained by a self trained network, I bounced upon the following error:

```
Could not detect contour
Index exceeds array bounds.

Error in CalculateStats (line 45)
stats.SignalToNoise = mean(1 - stats.Entropy(stats.ridgeTime));

Error in excel_Callback (line 34)
            stats =
            CalculateStats(I,windowsize,noverlap,nfft,rate,box,handles.settings.EntropyThreshold,handles.settings.AmplitudeThreshold);
            
Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('excel_Callback',hObject,eventdata,guidata(hObject)) 
Error while evaluating Menu Callback.
```
With older detection (.mat) files that I have obtained with networks inherent in DeepSqueak (e.g. All_Short_Calls_Network_V1.mat) the export function works fine. 

EDIT:
Also, I have been playing around with the files a bit and consequently found out that the aforementioned error occurs concurrent with the following error:

```
Error using matlab.graphics.axis.Axes/set
Error setting property 'CLim' of class 'Axes':
Value must be a 1x2 vector of numeric type in which the second element is larger than the first and may be Inf

Error in update_fig (line 17)
set(handles.axes1,'YDir', 'normal','YColor',[1 1 1],'XColor',[1 1 1],'Clim',[0 2*mean(max(I))]);

Error in DeepSqueak>NextCall_Callback (line 232)
    update_fig(hObject, eventdata, handles);

Error in DeepSqueak>figure1_WindowKeyPressFcn (line 329)
        NextCall_Callback(hObject, eventdata, handles)

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('figure1_WindowKeyPressFcn',hObject,eventdata,guidata(hObject)) 
Error while evaluating Figure WindowKeyPressFcn.
```
This error occurs when I am sliding through a detection file (.mat) and suddenly the slider gets stuck on a certain detected USV. However, by using the slider, I can skip the USV with problematic properties. Nonetheless, it is impossible to obtain the 'problem' USV on the GUI.
Finally, I will send an email with a .mat detection file that creates these errors. 
 
Kind regards,

Bavo Kempen

**Comment by DrCoffey** [comment]
Hey Bavo!

This one was easy. You had call at the exact end of the file and it was interfering with making the box. It is fixed now!

-Kevin

**Comment by BavoKempen** [comment]
Hey Kevin, 

I am glad that this one was easy :) Where do I have to change the code to make this work or will it be added later today in the master branch?

Kind regards,

Bavo

**Comment by BavoKempen** [comment]
Hey Kevin,

Yes indeed, I have been experimenting with the desktop app since yesterday and am watching a tutorial as we speak. 
Thanks again for the quick fix!

Kind regards,

Bavo

**Comment by younji74** [comment]
Hi, I have a similar error while trying to export call statistics in excel. The error msg is as following 

Reference to non-existent field 'settings'.

Error in excel_Callback (line 7)
[fname, fpath] = uigetfile([char(handles.settings.detectionfolder) '/*.mat'],'Select
Files to Export:','MultiSelect', 'on');

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 30)
    gui_mainfcn(gui_State, varargin{:});

Error in
matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('excel_Callback',hObject,eventdata,guidata(hObject)) 
Error while evaluating Menu Callback. 

Strangely, exporting into Raven file (txt) format works fine. So I can get the data that way. But we would love to have an excel file without converting the txt first. 

By the way, a great job!. We are using DeepSqueak to analyze anticipatory pleasure in rats. and it works really well despite the terrible ambient noise level. Thank you so much.

**Comment by MxMarx** [comment]
I've seen a few other people with a similar error. 

The line in the error message is different from the current version of "excel_Callback.m". A while ago, we refactored some code and moved a few function into new folders. I'd guess what happened is that there's either an older version of DeepSqueak in MATLAB's path, or DeepSqueak was updated and the old files are still there.

Removing and re-downloading everything should fix it. Other than that, I'm glad it's working for you!

-Russell
